lib/src/util.dart

```diff
@@ -96,6 +98,52 @@ String normalizeLinkLabel(String label) {
   return text;
 }
 
+///  Decodes HTML entity and numeric character references, for example decode
+/// `&#35` to `#`.
+String decodeHtmlCharacters(String input) =>
+    input.replaceAllMapped(htmlCharactersPattern, decodeHtmlCharacterFromMatch);
+
+/// Decodes HTML entity and numeric character references from the given [match].
+String decodeHtmlCharacterFromMatch(Match match) {
+  final text = match.match;
+  final isEntity = match[1] != null;
+  final isDecimal = match[2] != null;
+  final isHexadecimal = match[3] != null;
+  String? decodedText;
+
+  // Entity references, see
+  // https://spec.commonmark.org/0.30/#entity-references.
+  if (isEntity) {
+    decodedText = htmlEntitiesMap[text];
+  }
+
+  // Decimal numeric character references, see
+  // https://spec.commonmark.org/0.30/#decimal-numeric-character-references.
+  else if (isDecimal) {
+    final decimalValue = int.parse(match[2]!);
```

Comment:
The map access and null-check are still repeated here. We get `match[2]` above, and check if its null, then we might enter here and get `match[2]` again, and check if it is null, again.

It would be better to assign `match[2]` to a local variable and allow one null-check to promote.

Reply:
Thanks a lot! 

Fixed